Fix false positive

### Discussed in https://github.com/pyvista/pyvista/discussions/1474

<div type='discussions-op-text'>

<sup>Originally posted by **adeak** July  1, 2021</sup>
As of last night VTK 9.0.2. is released. Our tests seem to mostly run with them locally. One exception has already come up in CI:
```
tests/test_tinypages.py::test_tinypages FAILED                                                                                                                                                              [100%]

==================================================================================================== FAILURES =====================================================================================================
_________________________________________________________________________________________________ test_tinypages __________________________________________________________________________________________________

tmpdir = local('/tmp/pytest-of-adeak/pytest-221/test_tinypages0')

    @pytest.mark.skipif(os.name == 'nt', reason='path issues on Azure Windows CI')
    def test_tinypages(tmpdir):
        tmp_path = Path(tmpdir)
        html_dir = tmp_path / 'html'
        doctree_dir = tmp_path / 'doctrees'
        # Build the pages with warnings turned into errors
        cmd = [sys.executable, '-msphinx', '-W', '-b', 'html',
               '-d', str(doctree_dir),
               str(Path(__file__).parent / 'tinypages'), str(html_dir)]
        proc = Popen(cmd, stdout=PIPE, stderr=PIPE, universal_newlines=True,
                     env={**os.environ, "MPLBACKEND": ""})
        out, err = proc.communicate()
    
        assert proc.returncode == 0, \
            f"sphinx build failed with stdout:\n{out}\nstderr:\n{err}\n"
        if err:
>           pytest.fail(f"sphinx build emitted the following warnings:\n{err}")
E           Failed: sphinx build emitted the following warnings:
E           vtkDebugLeaks has found no leaks.

tests/test_tinypages.py:30: Failed
========================================================================================== memory consumption estimates ===========================================================================================
tests::test_tinypages.py::test_tinypages  - 226.5 MB
============================================================================================= short test summary info =============================================================================================
FAILED tests/test_tinypages.py::test_tinypages - Failed: sphinx build emitted the following warnings:
======================================================================================== 1 failed, 996 deselected in 8.75s ========================================================================================
vtkDebugLeaks has found no leaks.
```

Which seems like a false positive from a _good_ warning message being turned into an exception.

--- a/tests/test_tinypages.py
+++ b/tests/test_tinypages.py
@@ -26,8 +26,10 @@ def test_tinypages(tmpdir):
 
     assert proc.returncode == 0, \
         f"sphinx build failed with stdout:\n{out}\nstderr:\n{err}\n"
+
     if err:
-        pytest.fail(f"sphinx build emitted the following warnings:\n{err}")
+        if err.strip() != 'vtkDebugLeaks has found no leaks.':
+            pytest.fail(f"sphinx build emitted the following warnings:\n{err}")
 
     assert html_dir.is_dir()
 

